### PR TITLE
Copasi javascript support 

### PR DESCRIPTION
**Pull request by whomingbird**
https://github.com/seek4science/seek/issues/1939

### CODE REVIEW

File: app/assets/javascripts/copasi/copasi.js
Comment (stuzart) [inline app/assets/javascripts/copasi/copasi.js:1]: 3rd party so should be moved to `vendor/assets/javascript`

File: app/assets/javascripts/copasi/copasijs.js
Comment (stuzart) [inline app/assets/javascripts/copasi/copasijs.js:1]: 3rd party so should be moved to `vendor/assets/javascript`

File: app/assets/javascripts/plotly-2.27.0.min.js
Comment (stuzart) [inline app/assets/javascripts/plotly-2.27.0.min.js:1]: 3rd party so should be moved to `vendor/assets/javascript`

File: app/controllers/models_controller.rb [hunk 1]
```diff
@@ -129,6 +130,16 @@ def jws_enabled
     end
   end
 
+  def copasi_enabled
```
Comment (stuzart) [inline app/controllers/models_controller.rb:133]: shouldn't this be in `Seek::Copasi::Simulator` ?

File: app/helpers/assets_helper.rb [hunk 2]
```diff
@@ -214,15 +214,26 @@ def download_or_link_button(asset, download_path, link_url, _human_name = nil, o
...
+
+    auth_code = @model.special_auth_codes.where('code LIKE ?', 'copasi_%').first.code unless @model.can_download?(nil)
+
+    download_path = polymorphic_path([@model, blob], action: :download, code: auth_code)
 
-    files =   asset.content_blobs
-    download_path = polymorphic_path([files.first.asset, files.first], action: :download, code: params[:code])
     copasi_download_path =  "copasi://process?downloadUrl=http://"+request.host_with_port+download_path+"&activate=Time%20Course&createPlot=Concentrations%2C%20Volumes%2C%20and%20Global%20Quantity%20Values&runTask=Time-Course"
 
     tooltip_text_copasi_button = "Simulate the publicly accessible model in your local installed Copasi. "
```
Comment (stuzart) [inline app/helpers/assets_helper.rb:234]: does it have to be publicly accessible? 

`show_copasi_button?` suggests not.

File: app/helpers/models_helper.rb [hunk 3]
```diff
@@ -40,6 +40,6 @@ def jws_online_logo
   end
 
   def show_copasi_button?
-    Seek::Config.copasi_enabled && (@display_model.policy.access_type == Policy::ALL_USERS) && @display_model.is_copasi_supported?
+    Seek::Config.copasi_enabled && @display_model.is_copasi_supported? && @display_model.is_downloadable? && @display_model.can_download?(current_user)
```
Comment (stuzart) [inline app/helpers/models_helper.rb:43]: this duplicates and could reuse `@display_model.can_run_copasi?`

File: db/schema.rb
Comment (stuzart) [inline db/schema.rb:1]: the schema.rb shouldn't have changes, as there are no migrations.

**Review by stuzart — CHANGES_REQUESTED** [review]

File: app/helpers/models_helper.rb (hunk 3 above)
Comment (whomingbird) [reply app/helpers/models_helper.rb:43]: `can_run_copasi?` is defined for the "model" model, can not be called by @display_model ( versioning model)

**Review by stuzart — APPROVED** [review]

**Review by stuzart — APPROVED** [review]

**Pull request merged**